source/common/http/http2/codec_impl.cc

```diff
@@ -1644,15 +1610,6 @@ ConnectionImpl::Http2Callbacks::Http2Callbacks() {
         return static_cast<ConnectionImpl*>(user_data)->onSend(data, length);
       });
 
-  nghttp2_session_callbacks_set_send_data_callback(
```

Comment:
can you remind me why we no longer need this callback for nghttp2?  And the implications of removing DATA_FLAG_NO_COPY ?

Reply:
This callback became unused when we default-enabled the new QUICHE HTTP/2 codec wrapper API.

The wrapper uses a http2::adapter::DataFrameSource implementation when sending HTTP/2 DATA frames. This PR is removing the extra layers of translation between the nghttp2_data_provider and DataFrameSource data types. The actual DATA frame payload bytes still do not pass through the codec library, so no extra copies are introduced by dropping DATA_FLAG_NO_COPY.